### PR TITLE
Bump json-server from 0.10.1 to 0.17.4

### LINKED ISSUE #4576: Bump json-server from 0.10.1 to 0.17.4
#### What are you trying to achieve?

Bump json-server from 0.10.1 to 0.17.4 to get rid of some security issues. Seems to be the latest stable non release candidate.

### PR DESCRIPTION
**Pull request by thomashohn**
## Motivation/Description of the PR
- Removes some security related issues
- Resolves #4576

Applicable helpers:

- [ ] Playwright
- [ ] Puppeteer
- [ ] WebDriver
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] TestCafe

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] stepTimeout
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [ ] :bug: Bug fix
- [x] 🧹 Chore
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
   },
   "repository": "Codeception/codeceptjs",
   "scripts": {
-    "json-server": "./node_modules/json-server/bin/index.js test/data/rest/db.json -p 8010 --watch -m test/data/rest/headers.js",
+    "json-server": "npx json-server test/data/rest/db.json -p 8010 --watch -m test/data/rest/headers.js",
```
Comment (kobenguyent) [inline package.json]: I guess we don't need npx here.

Comment (thomashohn) [reply package.json]: The old command does not work - if you have a working one - please provide me with it

Comment (thomashohn) [reply package.json]: Ah - see it - will fix

**Comment by thomashohn** [comment]
@kobenguyent Any clue why the Acceptance Tests using docker compose / build (20.x) (pull_request) are failing ?

**Comment by kobenguyent** [comment]
> @kobenguyent Any clue why the Acceptance Tests using docker compose / build (20.x) (pull_request) are failing ?

I'm thinking of another alternative to this.

**Comment by thomashohn** [comment]
Found the error - new PR coming up

**Comment by kobenguyent** [comment]
> Found the error - new PR coming up

As you're looking into this. May you think this alternative is better as we are running tests inside containers https://testcontainers.com/?language=nodejs

**Pull request closed without merging**